src/cider/tests/utils/CiderBatchChecker.cpp

```diff
@@ -382,7 +382,7 @@ bool CiderBatchChecker::checkArrowEq(
   // This check is only a shortcut
   // even if it fails, it does not necessarily mean the results are wrong
   // we can goto step 4 and check the results row-by-row
-  if (expected_batches.size() == 1 && actual_batches.size() == 1) {
+  if (!ignore_order && expected_batches.size() == 1 && actual_batches.size() == 1) {
```

Comment:
ignore_order will make step3 fail but go through step4? so this is just for skipping useless steps?

Reply:
Yes, it won't affect correctness, but the logic here should be more precise.